pkg/containers/containers.go

```diff
@@ -28,20 +30,21 @@ type Containers struct {
 	cgroups  map[uint32]CgroupInfo
 	deleted  []uint64
 	mtx      sync.RWMutex // protecting both cgroups and deleted fields
+	enricher runtimeInfoService
 }
 
 // CgroupInfo represents a cgroup dir (might describe a container cgroup dir).
 type CgroupInfo struct {
-	Path        string
-	ContainerId string
-	Runtime     string
-	Ctime       time.Time
-	expiresAt   time.Time
+	Path      string
+	Container runtime.ContainerMetadata
+	Runtime   string
+	Ctime     time.Time
+	expiresAt time.Time
 }
 
 // InitContainers initializes a Containers object and returns a pointer to it.
 // User should further call "Populate" and iterate with Containers data.
-func InitContainers() (*Containers, error) {
+func InitContainers(sockets runtime.Sockets) (*Containers, error) {
```

Comment:
the stauts of `sockets` is supposed to change? if yes why is it passed by value? shouldn't it be passed by reference/pointer?

Reply:
I'm passing it by value because it's not supposed to change here, this is intentional. Sockets are initialized in CLI config time and injected downwards.